### PR TITLE
[#88] Tests for Discord ingestion

### LINKED ISSUE #88: [Epic 8] Tests for Discord ingestion
## Description

Write tests for jobs/discord.ts covering the core behaviour and edge cases.

## Acceptance Criteria

- [ ] Messages outside the weekly window are filtered out
- [ ] Bot messages are excluded
- [ ] Pagination works correctly — all messages across multiple pages are collected
- [ ] Rate limit responses (429) are retried after the correct wait time
- [ ] 401, 403, and 404 errors throw with a descriptive message
- [ ] Projects with no channels configured are skipped without error
- [ ] A failed channel fetch marks the sync job as FAILED with the error message recorded

## Dependencies/Blockers

#87

### PR DESCRIPTION
**Pull request by 14yao**
## Description
Adds full vitest test coverage for the Discord ingestion module.

## Solution
Tests are organised by exported function — `timestampToSnowflake`, `requestMessages`, and `runDiscordIngestion` — each as a top-level `describe` block. Covers all acceptance criteria:

- Messages outside the weekly window are filtered out
- Bot messages are excluded
- Pagination collects messages across multiple pages with correct cursor usage
- 429 responses retry after the wait time specified in the response body
- 401, 403, and 404 errors throw with descriptive messages
- Projects with no channels are skipped without creating a sync job
- Failed channel fetches mark the sync job as `FAILED` with the error message recorded

## Notes
- `vi.clearAllMocks()` runs in `beforeEach` to prevent state leaking between tests
- Fake timers are used for the 429 retry test to avoid real waits
- Shared `makeMessage`, `mockMessagesResponse`, and `mockErrorResponse` helpers keep tests concise

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,19 +41,17 @@ jobs:
...
           DIRECT_URL: ${{ secrets.DIRECT_URL }}
           SHADOW_DATABASE_URL: postgresql://postgres:postgres@localhost:5433/shadow
         run: |
           set +e
-          pnpm --filter @repo/db exec prisma migrate diff \
+          DIFF_OUTPUT=$(pnpm --filter @repo/db exec prisma migrate diff \
             --from-url "$DIRECT_URL" \
             --to-migrations prisma/migrations \
-            --shadow-database-url "$SHADOW_DATABASE_URL" \
-            --exit-code > /dev/null 2>&1
```
Comment (kelvinchen8) [inline .github/workflows/ci.yml:53]: doesnt dropping --exit-code break the drift detection? by default prisma migrate diff exits with code 0 whether or not a diff exists. it only returns non zero on internal error. so this will always report "no schema drift" unless the actual command fails

File: apps/worker/src/jobs/discord.test.ts [hunk 2]
```diff
@@ -1,115 +1,299 @@
...
+import { describe, it, expect, vi, beforeEach } from 'vitest'
 import { db } from '@repo/db'
 import { mockFetch } from '../test-config/vitest.setup'
-import { runDiscordIngestion } from './discord'
+import { runDiscordIngestion, requestMessages, timestampToSnowflake } from './discord'
 
-function mockMessagesResponse(messages: unknown[]) {
+function mockMessagesResponse(
+  messages: unknown[],
+  options: { remaining?: number; resetAfter?: number } = {}
```
Comment (kelvinchen8) [inline apps/worker/src/jobs/discord.test.ts]: is this being used right now? the 429 test creates its own response object.

**Review by kelvinchen8 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -38,19 +41,17 @@ jobs:
             echo "Waiting for shadow DB..."
             sleep 1
           done
-          sleep 2
```
Comment (kelvinchen8) [inline .github/workflows/ci.yml:41]: this was added as a safety buffer to prevent flaking tests but it should be ok to remove? we'll see, we can keep it removed for now

**Review by kelvinchen8 — APPROVED** [review]
lgtm
